### ISSUE TITLE
Bulk Editing Products Returns Blank Error [Fix inside]

### ISSUE THREAD

**Issue by fweb** [mailchimp/mc-magento#209]
Mailchimp for Magento 1.1.6
Magento CE 1.9.2.4

In admin backend, using bulk edts / actions in Catalog > Manage Products will yield a blank error on Save.

When a product is updated the extension incorrectly looks up the API key so it can push the changes upstream.

MailChimp uses a universally unique string to identify different merchants/stores communicating to it through the API. When the plugin tries to look up a Magento configuration value (such as the API key), it is supposed to turn the universal ID into a local store ID (which is just a small integer). It's not doing this conversion before asking Magento for the API key, and is instead supplying MailChimp's ID for the store. Magento has no idea what store that is, so this becomes the point of failure.

 **app/code/local/XXXXXXX/MailChimp/Model/Observer.php**

```
<?php

class XXXXXXX_MailChimp_Model_Observer extends Ebizmarts_MailChimp_Model_Observer {

    /**
     * We overrode this to supply the Magento store ID to update(), which is what it expects.
     *
     * @param Varien_Event_Observer $observer
     */
    public function productAttributeUpdate(Varien_Event_Observer $observer)
    {
        Mage::log(__METHOD__, null, 'observer.log', true);
        $productIds = $observer->getEvent()->getProductIds();
        $storeId = $observer->getEvent()->getStoreId();
        $scope = 'stores';
        if ($storeId === 0) {
            $scope = 'default';
        }

        //$mailchimpStoreId = Mage::helper('mailchimp')->getMCStoreId($storeId, $scope);
        Mage::log($storeId, null, 'observer.log', true);
        foreach ($productIds as $productId) {
            Mage::getModel('mailchimp/api_products')->update($productId, $storeId, null, null, 1, null, true);
        }
    }

}
```

**app/code/local/XXXXXXX/MailChimp/etc/config.xml**

```
<config>
    <modules>
        <XXXXXXX_MailChimp>
            <!-- keep this the same version as the ebizmarts module -->
            <version>1.1.6</version>
        </XXXXXXX_MailChimp>
    </modules>
    <global>
        <models>
            <mailchimp>
                <rewrite>
                    <observer>XXXXXXX_MailChimp_Model_Observer</observer>
                </rewrite>
            </mailchimp>
        </models>
    </global>
</config>
```

**Comment by yoosefi** [comment]
Just a note, the above fix is a custom module to override v1.1.6, and so needs an XML file to enable it.

`app/etc/modules/XXXXXXX_MailChimp.xml`

```
<config>
    <modules>
        <XXXXXXX_MailChimp>
            <active>true</active>
            <codePool>local</codePool>
        </XXXXXXX_MailChimp>
    </modules>
</config>
```

**Comment by Santiagoebizmarts** [comment]
Hello,

The fix is already in the module. You do not need to add the fix made by @fweb 
The commit that solves this is referenced above and it is https://github.com/mailchimp/mc-magento/commit/ea05c93c60a8f62ebbb82a6d651658219afb14bb

Regards
